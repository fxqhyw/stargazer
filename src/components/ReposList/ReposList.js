import React from 'react'
import { useSelector } from 'react-redux'
import { Table } from 'react-bootstrap'
import RepoItem from '../RepoItem/'
import { orderBy } from 'lodash'

const ReposList = () => {
  const repositories  = useSelector(state => state.repositories.data)
  
  return(
    <Table>
      <thead>
        <tr>
          <th>Repository name</th>
          <th>Stars</th>
          <th></th>
        </tr>
      </thead>
      <tbody>
        {orderBy(Object.values(repositories), ['stargazers_count'], ['desc']).map(repo =>
          <RepoItem
            key={repo.id}
            id={repo.id}
            name={repo.full_name}
            starsCount={repo.stargazers_count}
          />
        )}
      </tbody>
    </Table>
  )
}

export default ReposList;