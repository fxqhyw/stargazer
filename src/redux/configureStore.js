import { createStore, applyMiddleware } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly'
import { createLogicMiddleware } from 'redux-logic'
import Reducer from './reducers'
import rootLogic from './logic/'
import axios from 'axios'

const initialState = {
  repositories: {
    data: {},
    error: null
  }
}

const logicDeps = {
    httpClient: axios
}

const logicMiddleware = createLogicMiddleware(rootLogic, logicDeps);


export default function configureStore() {
  return createStore(
    Reducer,
    initialState,
    composeWithDevTools(
      applyMiddleware(logicMiddleware),
    )
  )
}

