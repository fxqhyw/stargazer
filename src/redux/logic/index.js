import reposDtataHandingLogic from './reposDataHandlingLogic'

export default [
  ...reposDtataHandingLogic
]
