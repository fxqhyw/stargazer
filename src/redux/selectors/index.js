import { createSelector } from 'reselect'
import { orderBy } from 'lodash'

const repositories = state => state.repositories

export const repositoryError = createSelector(
  repositories,
  repositories => repositories.error
)

export const repositoriesData = createSelector(
  repositories,
  repositories => orderBy(Object.values(repositories.data), ['stargazers_count'], ['desc'])
)


export const repositoryData = (id) => createSelector(
  repositories,
  repositories => repositories.data[id]
)
