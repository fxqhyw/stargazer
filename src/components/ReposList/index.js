import ReposList from './component'

export default ReposList
