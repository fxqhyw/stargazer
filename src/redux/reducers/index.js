import { combineReducers } from 'redux' 
import reposDataHandling from './reposDataHandling'

export default combineReducers({
  repositories: reposDataHandling
})
