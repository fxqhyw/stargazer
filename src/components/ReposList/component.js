import React from 'react'
import { useSelector } from 'react-redux'
import { Table } from 'react-bootstrap'
import RepoItem from '../RepoItem'
import { repositoriesData } from '../../redux/selectors'

const ReposList = () => {
  const repositories  = useSelector(repositoriesData)
  
  return(
    <Table>
      <thead>
        <tr>
          <th>Repository name</th>
          <th>Stars</th>
          <th></th>
        </tr>
      </thead>
      <tbody>
        {repositories.map(repo =>
          <RepoItem
            key={repo.id}
            id={repo.id}
            name={repo.full_name}
            starsCount={repo.stargazers_count}
          />
        )}
      </tbody>
    </Table>
  )
}

export default ReposList
