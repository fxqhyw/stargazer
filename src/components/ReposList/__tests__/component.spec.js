import React from 'react'
import { shallow } from 'enzyme'
import ReposList from '../component'
import { repositoriesData } from '../../../redux/selectors'


jest.mock('../../../redux/selectors', () => ({
  repositoriesData: jest.fn()
}))

jest.mock('react-redux', () => ({
  useSelector: jest.fn(fn => fn()),
}))
 
describe('RepoList', () => {
  let component;
 
  beforeEach(() => {
    repositoriesData.mockReturnValue([
      {
        id: 10270250,
        full_name: "facebook/react",
        stargazers_count: 154386,
        subscribers_count: 6678,
        forks_count: 30270,
        clone_url: "https://github.com/facebook/react.git",
        languages_url: "https://api.github.com/repos/facebook/react/languages"
      },
      {
        id: 11730342,
        full_name: "vuejs/vue",
        stargazers_count: 170383,
        subscribers_count: 6263,
        forks_count: 26098,
        clone_url: "https://github.com/vuejs/vue.git",
        languages_url: "https://api.github.com/repos/vuejs/vue/languages"
      }
    ])
  })
 
  it('should render correctly', () => {
    component = shallow(
      <ReposList />
    )
    expect(component).toMatchSnapshot()
  });

  afterEach(() => {
    component.unmount()
  })
});
