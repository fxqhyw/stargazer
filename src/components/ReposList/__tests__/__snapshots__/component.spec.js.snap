// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoList should render correctly 1`] = `
<ForwardRef>
  <thead>
    <tr>
      <th>
        Repository name
      </th>
      <th>
        Stars
      </th>
      <th />
    </tr>
  </thead>
  <tbody>
    <RepoItem
      id={10270250}
      key="10270250"
      name="facebook/react"
      starsCount={154386}
    />
    <RepoItem
      id={11730342}
      key="11730342"
      name="vuejs/vue"
      starsCount={170383}
    />
  </tbody>
</ForwardRef>
`;
